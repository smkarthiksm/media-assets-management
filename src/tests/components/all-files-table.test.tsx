import '@testing-library/jest-dom';
import { renderWithProviders } from '../test-utility';
import { fireEvent, screen } from '@testing-library/react';
import AllFilesTableComponent from '../../components/all-files-table/all-files-table';
import { allFilesStateStub } from '../stubs/all-files-state.stub';

describe('AllFilesTableComponent', () => {
  test('should render', () => {
    const { container } = renderWithProviders(<AllFilesTableComponent />);
    expect(container).toMatchSnapshot();
  });

  test('should render empty table when no data is present', () => {
    renderWithProviders(<AllFilesTableComponent />, {
      preloadedState: {
        allFiles: { allFiles: [], cachedAllFiles: [], searchValue: '' },
      },
    });

    const element = screen.getByRole('cell') as HTMLTableCellElement;
    expect(element.innerHTML).toEqual('No records found');
  });

  test('should render table when data is present', () => {
    renderWithProviders(<AllFilesTableComponent />, {
      preloadedState: {
        allFiles: allFilesStateStub,
      },
    });
    const elements = screen.getAllByRole('row') as HTMLTableRowElement[];
    expect(elements).toHaveLength(3);
  });

  test('should populate first row fields and open edit modal', () => {
    const { store } = renderWithProviders(<AllFilesTableComponent />, {
      preloadedState: {
        allFiles: allFilesStateStub,
      },
    });
    const editButton = screen.getAllByTestId('EditIcon')[0];
    fireEvent.click(editButton);
    expect(store.getState().editFileModal).toEqual({
      title: 'testTitle1',
      album: 'testAlbum1',
      artist: 'testArtist1',
      index: 0,
      isEditFileModalVisible: true,
    });
  });

  test('should populate first row fields and open delete modal', () => {
    const { store } = renderWithProviders(<AllFilesTableComponent />, {
      preloadedState: {
        allFiles: allFilesStateStub,
      },
    });
    const deleteButton = screen.getAllByTestId('DeleteIcon')[0];
    fireEvent.click(deleteButton);
    expect(store.getState().deleteFileModal).toEqual({
      index: 0,
      isDeleteFileModalVisible: true,
    });
  });
});
