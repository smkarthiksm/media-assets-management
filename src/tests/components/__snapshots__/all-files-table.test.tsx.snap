// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllFilesTableComponent should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            #
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Title
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Album
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Artist
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
            scope="col"
          >
            Duration
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            File type
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
            colspan="7"
          >
            No records found
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
